File: scala/src/main/scala/com/pduda/method/ClientOfAuthenticator.scala
package com.pduda.method

object ClientOfAuthenticator {
  def main(args: Array[String]): Unit =
    new ClientOfAuthenticator(new AuthenticationService).run()
}
class ClientOfAuthenticator(private var authenticationService: AuthenticationService) {
  def run(): Unit = {
    val authenticated = authenticationService.isAuthenticated(33)
    println("33 is authenticated = " + authenticated)
  }
}

File: Java/src/main/java/com/pduda/method/ClientOfAuthenticator.java
package com.pduda.method;

import java.util.logging.Level;
import java.util.logging.Logger;

public class ClientOfAuthenticator {

    private static final Logger LOGGER = Logger.getLogger(ClientOfAuthenticator.class.getName());
    private AuthenticationService authenticationService;

    public static void main(String[] args) {
        new ClientOfAuthenticator(new AuthenticationService()).run();
    }

    public ClientOfAuthenticator(AuthenticationService authenticationService) {
        this.authenticationService = authenticationService;
    }

    public void run() {
        boolean authenticated = authenticationService.isAuthenticated(33);
        LOGGER.log(Level.INFO, "33 is authenticated = {0}", authenticated);
    }
}
